util/reggen/gen_dv.py

```diff
@@ -72,7 +104,44 @@ def gen_core_file(outdir: str,
         yaml.dump(core_data, core_file, encoding='utf-8')
 
 
-def gen_dv(block: IpBlock, dv_base_prefix: str, outdir: str) -> int:
+def get_dv_base_names_objects(dv_base_names: List[str]) -> Dict[str, DvBaseNames]:
+    '''Convert the dv_base_names from string to a list of DvBaseNames objects.
+
+    If the `dv_base_names` matches the default argument, returns default DvBaseNames object.
+    If the `dv_bave_names` has customerized input, it should follow this format of a list:
+    ['block-1:type:entity-name', 'block-2:type:entity-name',...].
+    This method will parse the list and return customerized DvBaseNames objects.
+    '''
```

Comment:
How about the following:
```suggestion
    '''Returns a dictionary mapping a DvBaseNames object to a block.

    `dv_base_names` is a list of base class entity names provided on the command-line,
    in the following format:
        ast:block:ast_base_reg_block ast:pkg:ast_base_reg_pkg otp_ctrl:all:otp_ctrl_base
      
    This function creates a dictionary that wraps the provided base class overrides for each 
    block within a `DvBaseNames` object and returns a dictionary mapping the object to the 
    block. 
    '''
```

Note that with `nargs=+`, the list on the command line is space-separated.

Reply:
Thanks Sri, yeah I was thinking to show it better as a list, but I think your wording is better for sure :) I will update that. Thanks again.